src/components/DragAndDrop/Sortable.tsx

```diff
@@ -46,10 +49,22 @@ export const shouldStack = (id: UniqueIdentifier, items: UniqueIdentifier[], new
 export const Sortable = ({id, children, disabled, className, columnId, setItems}: SortableProps) => {
   const dispatch = useAppDispatch();
   const {collisions} = useDndContext();
+
+  // Check if this note is locked by another user
+  const {lockedNotes} = useAppSelector((state) => state.dragLocks);
+  const currentUserId = useAppSelector((state) => state.participants?.self?.user?.id);
+  const participants = useAppSelector((state) => state.participants);
+  const isLockedByOther = lockedNotes[id.toString()] && lockedNotes[id.toString()] !== currentUserId;
+  const dragDisabled = disabled || isLockedByOther;
+
+  // Get the user who is dragging this note
+  const draggingUserId = lockedNotes[id.toString()];
+  const draggingUser = draggingUserId ? [...(participants?.others || []), participants?.self].find((p) => p?.user?.id === draggingUserId) : null;
```

Comment:
this is a little messy:
1. `dragDisabled` is of type `boolean | ""`
2. `draggingUser` is of type `Participant | undefined | null`
3. L62 (`draggingUser = ...`) is hard to read

Maybe consider using [non-null assertions](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-2-0.html#non-null-assertion-operator) for the state selectors, as on a board, we can safely assume users and participants do exist.